acstools/acsphotcte.py

```diff
@@ -0,0 +1,289 @@
+data, please see Chiaberge et al. 2009 (ISR 2009-01), or use pixel-based
+CTE-corrected files obtained from MAST. Currently,  only 3 and 5
+pixel aperture radii are supported. The model is designed to compute the
+CTE losses as a function of the number of Y transfers, sky background,
+source flux, and observation date.
+
+
+Usage Guidelines
+----------------
+As this service is hosted on AWS, there are some guidelines that should be
+followed when utilizing this tool. For each call to
+:py:meth:`PhotCTEAPI.correct_photomerty`, AWS will assemble the compute resources,
+install the software, compute the CTE corrections, send the result back,
+and then terminate all resources. Hence, users should try to process as many
+sources as they can in a single function call. Testing has shown that the
+optimal number is < 25000, but the service can handle as many as 150,000
+sources in a single request.
+
+
+Example
+-------
+Get the CTE corrected FLT photometry for a list of 1000 sources:
+
+>>> import numpy as np
+>>> from acstools import acsphotcte
+>>> ytransfers = 2048 * np.random.random(size=1000)
+>>> fluxes = 20000*np.random.random(size=len(ytransfers))
+>>> local_skys = 80*np.random.random(size=len(ytransfers))
+>>> mjd = 59341.
+>>> radius = 3.
+>>> cte_corrections = acsphotcte.correct_photometry(
+...    radius=radius,
+...    ytransfers=ytransfers,
+...    mjd=mjd,
+...    local_skys=local_skys,
+...    fluxes=fluxes
+...)
+>>> print(cte_corrections)
+"""
+from collections import Iterable
+import json
+import logging
+
+import numpy as np
+import requests
+logging.basicConfig(format='%(levelname)-4s '
+                           '[%(module)s.%(funcName)s:%(lineno)d]'
+                           ' %(message)s',
+                    )
+LOG = logging.getLogger('PhotCTEAPI')
+LOG.setLevel(logging.INFO)
+
+__taskname__ = "acsphotcte"
+__author__ = "Nathan Miles"
+__version__ = "1.0"
+__vdate__ = "16-Mar-2020"
+
+__all__ = ['PhotCTEAPI']
+
+
+class PhotCTEAPI():
+    """Convenience class for handling queries to ACS Photometric CTE API
+    """
+    def __init__(self):
+
+        # API Endpoint for the ACS Photometric CTE API
+        self._api_url = 'https://hbview9n7f.execute-api.us-east-1' \
+                        '.amazonaws.com/dev/run-photcte-correction'
+        # Public API ID for the ACS Photometric CTE API
+        self._api_id = ''
+        # Public API key for the ACS PHotometric CTE API
+        self._api_key = ''
+        self._cte_corrections = None
+        self._corrected_fluxes = None
+
+    @property
+    def corrected_fluxes(self):
+        """CTE corrected fluxes"""
+        return self._corrected_fluxes
+
+    @corrected_fluxes.setter
+    def corrected_fluxes(self, value):
+        self._corrected_fluxes = value
+
+    @property
+    def cte_corrections(self):
+        """Computed CTE corrections"""
+        return self._cte_corrections
+
+    @cte_corrections.setter
+    def cte_corrections(self, value):
```

Comment:
Is there no need for value validation here?

Reply:
Same as above